frontend/src/app/modules/grids/widgets/wp-table/wp-table.component.ts

```diff
@@ -0,0 +1,146 @@
+import {Component, OnInit, OnDestroy, ViewChild, AfterViewInit} from "@angular/core";
+import {WidgetWpListComponent} from "core-app/modules/grids/widgets/wp-widget/wp-widget.component";
+import {WorkPackageTableConfiguration} from "core-components/wp-table/wp-table-configuration";
+import {QueryResource} from "core-app/modules/hal/resources/query-resource";
+import {I18nService} from "core-app/modules/common/i18n/i18n.service";
+import {untilComponentDestroyed} from 'ng2-rx-componentdestroyed';
+import {WorkPackageIsolatedQuerySpaceDirective} from "core-app/modules/work_packages/query-space/wp-isolated-query-space.directive";
+import {skip, take} from 'rxjs/operators';
+import {UrlParamsHelperService} from "core-components/wp-query/url-params-helper";
+import {QueryFormDmService} from "core-app/modules/hal/dm-services/query-form-dm.service";
+import {QueryDmService} from "core-app/modules/hal/dm-services/query-dm.service";
+import {QueryFormResource} from "core-app/modules/hal/resources/query-form-resource";
+
+@Component({
+  templateUrl: './wp-table.component.html',
+  styleUrls: ['../wp-widget/wp-widget.component.css']
+})
+export class WidgetWpTableComponent extends WidgetWpListComponent implements OnInit, OnDestroy, AfterViewInit {
+  public text = { title: this.i18n.t('js.grid.widgets.work_packages_table.title') };
+  public queryId:string|null;
+  private queryForm:QueryFormResource;
+  public inFlight = false;
+  public query:QueryResource;
+
+  public configuration:Partial<WorkPackageTableConfiguration> = {
+    actionsColumnEnabled: false,
+    columnMenuEnabled: true,
+    hierarchyToggleEnabled: true,
+    contextMenuEnabled: false
+  };
+
+  constructor(protected i18n:I18nService,
+              protected urlParamsHelper:UrlParamsHelperService,
+              private readonly queryDm:QueryDmService,
+              private readonly queryFormDm:QueryFormDmService) {
+    super(i18n);
+  }
+
+  @ViewChild(WorkPackageIsolatedQuerySpaceDirective) public querySpaceDirective:WorkPackageIsolatedQuerySpaceDirective;
+
+  ngOnInit() {
+    if (!this.resource.options.queryId) {
+      this.createInitial()
+        .then((query) => {
+          this.resource.options = { queryId: query.id };
+
+          this.resourceChanged.emit(this.resource);
+
+          this.queryId = query.id;
+
+          super.ngOnInit();
+        });
+    } else {
+      this.queryId = this.resource.options.queryId as string;
+
+      super.ngOnInit();
+    }
+  }
+
+  ngAfterViewInit() {
+    this
+      .querySpaceDirective
+      .querySpace
+      .query
+      .values$()
+      .pipe(
+        take(1),
+        untilComponentDestroyed(this)
+      ).subscribe((query) => {
+        this.query = query;
+        this.queryId = query.id;
+      });
+
+    this
+      .querySpaceDirective
+      .querySpace
```

Comment:
I would have preferred wrapping this directive in one that already has `isolated-query-space` so you can natively inject the querySpace since it avoids needing to access the ViewChild. Anyhow, willing to merge this as is.